packages/db/src/lib/granule.ts

```diff
@@ -140,6 +147,27 @@ export const getUniqueGranuleByGranuleId = async (
   return PgGranuleRecords[0];
 };
 
+/**
+ * Get one Granule for a granule_id and collection_cumulus_id
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction -
+ *  DB client or transaction
+ * @param {string} granuleId - a granule.granule_id
+ * @param {string} collectionCumulusId - a granule.collection_cumulus_id
+ * @param {GranulePgModel} granulePgModel - Granule PG model class instance
+ * @returns {Promise<PostgresGranuleRecord>}
+ */
+export const getGranuleByUniqueColumns = (
```

Comment:
Also - why is this method needed at all?     Given we're doing the collection string -> cumulus_collection_id (probably correctly) logic in the api endpoint, all this is doing is formatting from a explicit keyed parameter block to a model.get formatted one, and then passing through to a model.get call which has required arguments validation.    

If the issue is not liking the `params` format on the `get` call, could we just change it on the model? 

If we do keep it, I think we should just call it.... getGranule?

Reply:
Yeah, it's not really necessary to have a lib function here wrapping the model `get`. I was thinking the "correct" pattern was to have a library method wrapping model calls but that's not right at all. We use library functions if we need some added logic we don't want in the model but otherwise calling the model directly is much clearer (and what we're doing all over the place). Sold.

https://github.com/nasa/cumulus/pull/2978/commits/8834e69ef2d567307dd546822dfe2888fa3579a5

Reply:
A week later and I remembered why I made the lib function originally 😆. We need exactly the same logic to fetch a granule by collectionId and granuleId in the `del()` method. I'll leave it like it is in this PR but it may be refactored back to an external function in a subsequent PR.

Reply:
YAGNI until the next PR it is.